jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryClientTest.java

```diff
@@ -211,6 +211,29 @@ public void testConfigureBasicAuth()
         registry.getInputRead(), CoreMatchers.containsString("Authorization: Basic dXNlcjpwYXNz"));
   }
 
+  @Test
+  public void testPullManifest()
+      throws IOException, InterruptedException, GeneralSecurityException, URISyntaxException,
+          RegistryException {
+    String imageTag = "testIMage";
+    String manifestResponse =
+        "HTTP/1.1 200 OK\nContent-Length: 307\n\n{\n"
+            + "    \"schemaVersion\": 2,\n"
+            + "    \"mediaType\": \"application/vnd.docker.distribution.manifest.v2+json\",\n"
+            + "    \"config\": {\n"
+            + "        \"mediaType\": \"application/vnd.docker.container.image.v1+json\",\n"
+            + "        \"size\": 7023,\n"
+            + "        \"digest\": \"sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7\"\n"
+            + "    }\n"
+            + "}";
+
+    registry = new TestWebServer(false, Arrays.asList(manifestResponse), 1);
+    RegistryClient registryClient = createRegistryClient(null);
+
+    ManifestAndDigest<?> manifestAndDigest = registryClient.pullManifest(imageTag);
+    Assert.assertEquals(66, manifestAndDigest.getManifest().toString().length());
```

Comment:
instead of this, let's verify more details with the returned manifest (`getManifest()`) + digest (`getDigest()`).

The manifest in this case would be of type `V22ManifestTemplate`. So you could
```java
Assert.assertTrue(manifestAndDigest.getManifest() instanceof V22ManifestTemplate);
V22ManifestTemplate manifest = (V22ManifestTemplate) manifestAndDigest.getManifest();
Assert.assertEquals(2, manifest.getSchemaVersion());
Assert.assertEquals(7023, manifest.getContainerConfiguration().getSize());
... // you can do more checks
```

Likewise, you'll be able to verify values in the returned digest.
```java
assertEquaus("sha256:...", manifest.getDigest().toString());
...
```

Reply:
This is a good suggestion.This enables us to confirm that this is the actual manifest we intended to receive.

Reply:
Exactly!